kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -223,6 +223,7 @@ private void forwardShortCircuitResponse(ResponseHeaderData header, ApiMessage r
                     "Attempt to respond with ApiMessage of type " + ApiKeys.forId(response.apiKey()) + " but request is of type " + decodedFrame.apiKey());
         }
         DecodedResponseFrame<?> responseFrame = new DecodedResponseFrame<>(decodedFrame.apiVersion(), decodedFrame.correlationId(), header, response);
+        decodedFrame.transferBuffers(responseFrame);
```

Comment:
Maybe these two lines would be better as `decodedFrame.copy(header, response)` but it's back to Generics hell since DecodedResponseFrame has a generic type B for the ApiMessage, but here `decodedFrame` is `DecodedFrame<?, ?>` so we don't know the ApiMessage type it's dealing with. It could be done as in the other PR, remove the generics,and work in terms of ApiMessage with additional checks to ensure we only copy in the same concrete ApiMessage type.

Reply:
Maybe best to get this fix in as is and look at refactoring the generics in another PR

Reply:
I think we should get this in and refactor later.